src/main/java/io/cryostat/net/web/http/api/v2/TargetsPostHandler.java

```diff
@@ -156,10 +159,23 @@ public IntermediateResponse<ServiceRef> handle(RequestParameters params) throws
                     throw new ApiException(400, "Duplicate connectUrl");
                 }
             }
-            Map<AnnotationKey, String> cryostatAnnotations = new HashMap<>();
 
-            String jvmId = jvmIdHelper.getJvmId(uri.toString());
+            String username = attrs.get("username");
+            String password = attrs.get("password");
+            Optional<Credentials> credentials =
+                    StringUtils.isBlank(username) || StringUtils.isBlank(password)
```

Comment:
I wonder if the target JVM would actually accept a `jmxremote.password` file defining a user with an empty password. Maybe we need to be more lenient for that case. Something to investigate anyway, out of scope of this particular PR. It could be tested by modifying Cryostat's own `entrypoint.sh` something like this:

```diff
diff --git a/src/main/extras/app/entrypoint.bash b/src/main/extras/app/entrypoint.bash
index 185d0340..4f9cd524 100755
--- a/src/main/extras/app/entrypoint.bash
+++ b/src/main/extras/app/entrypoint.bash
@@ -25,8 +25,10 @@ function createJmxCredentials() {
     fi
 
     echo -n "$CRYOSTAT_RJMX_USER $CRYOSTAT_RJMX_PASS" > "$PWFILE"
+    echo -n "NOPASS " >> "$PWFILE"
     chmod 400 "$PWFILE"
     echo -n "$CRYOSTAT_RJMX_USER readwrite" > "$USRFILE"
+    echo -n "NOPASS readwrite" >> "$USRFILE"
     chmod 400 "$USRFILE"
 }
```

but then there would also be required changes to the Credentials API and/or frontend to allow blank passwords to be stored.

Reply:
I open #1270 to track this :D